src/functions/cancellations-feed-validator/index.ts

```diff
@@ -0,0 +1,150 @@
+        );
+    }
+
+    const subscriptionBody: CancellationsSubscribeMessage = {
+        dataProducerEndpoint: subscription.url,
+        description: subscription.description,
+        shortDescription: subscription.shortDescription,
+        username: subscriptionUsername,
+        password: subscriptionPassword,
+        requestorRef: subscription.requestorRef ?? null,
+        subscriptionId: subscription.PK,
+        publisherId: subscription.publisherId,
+    };
+
+    const cancellationsProducerApiKey = await getSecret<string>({ SecretId: cancellationsProducerApiKeyArn });
+
+    await axios.post(subscribeEndpoint, subscriptionBody, { headers: { "x-api-key": cancellationsProducerApiKey } });
+};
+
+export const handler: Handler = async (event, context) => {
+    withLambdaRequestTracker(event ?? {}, context ?? {});
+
+    try {
+        logger.info("Starting cancellations feed validator");
+
+        const currentTime = getDate();
+
+        const {
+            TABLE_NAME: tableName,
+            SUBSCRIBE_ENDPOINT: subscribeEndpoint,
+            CANCELLATIONS_PRODUCER_API_KEY_ARN: cancellationsProducerApiKeyArn,
+        } = process.env;
+
+        if (!tableName || !subscribeEndpoint || !cancellationsProducerApiKeyArn) {
+            throw new Error(
+                "Missing env vars: TABLE_NAME, SUBSCRIBE_ENDPOINT and CANCELLATIONS_PRODUCER_API_KEY_ARN must be set",
+            );
+        }
+
+        const subscriptions = await getCancellationsSubscriptions(tableName);
+        const nonTerminatedSubscriptions = subscriptions.filter((subscription) => subscription.status !== "inactive");
+
+        if (!nonTerminatedSubscriptions) {
+            logger.info("No subscriptions found in DynamoDb to validate");
+            return;
+        }
+
+        await Promise.all(
+            nonTerminatedSubscriptions.map(async (subscription) => {
+                const subscriptionIsHealthy = checkCancellationsSubscriptionIsHealthy(subscription, currentTime);
+
+                if (subscriptionIsHealthy) {
+                    if (subscription.status !== "live") {
+                        await putDynamoItem<CancellationsSubscription>(tableName, subscription.PK, "SUBSCRIPTION", {
+                            ...subscription,
+                            status: "live",
+                        });
+                    }
+
+                    return;
+                }
+
+                await putDynamoItem<CancellationsSubscription>(tableName, subscription.PK, "SUBSCRIPTION", {
+                    ...subscription,
+                    status: "error",
+                });
+
+                try {
+                    await sendTerminateSubscriptionRequest(
+                        "cancellations",
+                        subscription.PK,
+                        subscription,
+                        isPrivateAddress(subscription.url),
```

Comment:
```suggestion
                        subscription.url,
```

I believe this function was used for the stagecoach site-to-site VPN connection - when we were discussing the tickets with LJ we decided not to build out cancellations to handle a private connection for now